frontend/src/app-components/inputs/Selectable.tsx

```diff
@@ -205,6 +209,22 @@ const Selectable: FC<SelectableProps> = ({
         onChange={(e) => handleTextChange(e.target.value)}
         placeholder={placeholder}
```

Comment:
```suggestion
        placeholder={placeholder}
        InputProps={{
        endAdornment: loading ? (
          <InputAdornment position="end">
            <CircularProgress size={20} />
          </InputAdornment>
        ) : null,
      }}
```
I think you can add it as a adornment to the input.

Reply:
Okay I will do that

Reply:
In the `TextField` component, we can use `InputProps` to pass the `endAdornment`, but in the `Input` component, we need to use `endAdornment` directly. While using `InputAdornment` with `TextField` works as expected, it did not behave as intended when applied to `Input`.  

To resolve this, I implemented custom CSS to properly position the spinner within the input field. After thorough testing across different devices, the solution is responsive and working as expected.

Reply:
Hello @marrouchi , 
Any further changes needed ?